### PR TITLE
Tests for questions with shared source only

### PR DESCRIPTION
**Pull request by SethPoulsen**
Implements #10861

### CODE REVIEW

**Comment by SethPoulsen** [comment]
> All modified and coverable lines are covered by tests ✅

Nice. Every change in the PR is in a test file.

File: apps/prairielearn/src/tests/sharedQuestionPreview.test.ts [hunk 1]
```diff
@@ -73,7 +72,7 @@ describe('Shared Question Preview', function () {
     await syncUtil.writeAndSyncCourseData(consumingCourseData);
   });
 
-  describe('Public Question Previews', function () {
+  describe('Public Question Previews', async function () {
```
Comment (nwalters512) [inline apps/prairielearn/src/tests/sharedQuestionPreview.test.ts:75]: This should not need to be async.

File: apps/prairielearn/src/tests/sharedQuestionPreview.test.ts [hunk 2]
```diff
@@ -82,11 +81,32 @@ describe('Shared Question Preview', function () {
       isStudentPage: false,
     };
 
-    testQuestionPreviews(previewPageInfo, addNumbers, addVectors);
-
-    testFileDownloads(previewPageInfo, downloadFile, false);
-
-    testElementClientFiles(previewPageInfo, customElement);
+    describe('When questions are share_source_publicly but not shared_publicly', async function () {
```
Comment (nwalters512) [inline apps/prairielearn/src/tests/sharedQuestionPreview.test.ts:84]: Ditto, not async. Same applies to the other `describe(...)` block below.

File: apps/prairielearn/src/tests/sharedQuestionPreview.test.ts [hunk 3]
```diff
@@ -82,11 +81,32 @@ describe('Shared Question Preview', function () {
...
+      before(
+        'Make sure question have share_source_publicly set but not shared_pubilcly',
+        async function () {
```
Comment (nwalters512) [inline apps/prairielearn/src/tests/sharedQuestionPreview.test.ts:87]: Nit: typically we'd reach for arrow functions here unless access to `this` is specifically needed. That said, I imagine you're just matching against other tests in this same file.

File: apps/prairielearn/src/tests/sharedQuestionPreview.test.ts [hunk 4]
```diff
@@ -82,11 +81,32 @@ describe('Shared Question Preview', function () {
...
     };
 
-    testQuestionPreviews(previewPageInfo, addNumbers, addVectors);
-
-    testFileDownloads(previewPageInfo, downloadFile, false);
-
-    testElementClientFiles(previewPageInfo, customElement);
+    describe('When questions are share_source_publicly but not shared_publicly', async function () {
+      before(
+        'Make sure question have share_source_publicly set but not shared_pubilcly',
```
Comment (nwalters512) [inline apps/prairielearn/src/tests/sharedQuestionPreview.test.ts:86]: "... question have ..." is improper grammar. "pubilcly" is spelled wrong.

Comment (SethPoulsen) [reply apps/prairielearn/src/tests/sharedQuestionPreview.test.ts]: fixed

File: apps/prairielearn/src/tests/sharedQuestionPreview.test.ts (hunk 3 above)
Comment (SethPoulsen) [reply apps/prairielearn/src/tests/sharedQuestionPreview.test.ts]: I just went ahead and updated the whole file, it was pretty simple.

File: apps/prairielearn/src/tests/sharedQuestionPreview.test.ts (hunk 2 above)
Comment (SethPoulsen) [reply apps/prairielearn/src/tests/sharedQuestionPreview.test.ts]: fixed

File: apps/prairielearn/src/tests/sharedQuestionPreview.test.ts (hunk 1 above)
Comment (SethPoulsen) [reply apps/prairielearn/src/tests/sharedQuestionPreview.test.ts]: fixed

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**